src/System.Memory/tests/Span/Overflow.cs

```diff
@@ -60,6 +60,219 @@ public static void IndexOverflow()
             }
         }
 
+        [Fact]
+        public static void SliceStartInt32Overflow()
+        {
+            IntPtr huge;
+            try
+            {
+                huge = Marshal.AllocHGlobal(new IntPtr(ThreeGiB));
+            }
+            catch (Exception)
+            {
+                return;  // It's not implausible to believe that a 3gb allocation will fail - if so, skip this test to avoid unnecessary test flakiness.
+            }
+
+            try
+            {
+                unsafe
+                {
+                    int
+                        GuidThreeGiBLimit = (int)(ThreeGiB / sizeof(Guid)),
+                        GuidTwoGiBLimit = (int)(TwoGiB / sizeof(Guid)),
+                        GuidOneGiBLimit = (int)(OneGiB / sizeof(Guid));
+
+                    var span = new Span<Guid>((void*)huge, GuidThreeGiBLimit);
+                    Guid guid = Guid.NewGuid();
+                    var slice = span.Slice(GuidTwoGiBLimit + 1);
+                    slice[0] = guid;
+                    slice = span.Slice(GuidOneGiBLimit).Slice(1).Slice(GuidOneGiBLimit);
+                    Assert.Equal(guid, slice[0]);
+                }
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(huge);
+            }
+        }
+
+        [Fact]
+        public static void SliceStartUInt32Overflow()
+        {
+            IntPtr huge;
+            try
+            {
+                huge = Marshal.AllocHGlobal(new IntPtr(ThreeGiB));
+            }
+            catch (Exception)
+            {
+                return;  // It's not implausible to believe that a 3gb allocation will fail - if so, skip this test to avoid unnecessary test flakiness.
+            }
+
+            try
+            {
+                unsafe
+                {
+                    int
+                        GuidThreeGiBLimit = (int)(ThreeGiB / sizeof(Guid)),
+                        GuidTwoGiBLimit = (int)(TwoGiB / sizeof(Guid)),
+                        GuidOneGiBLimit = (int)(OneGiB / sizeof(Guid));
+
+                    var span = new Span<Guid>((void*)huge, GuidThreeGiBLimit);
+                    Guid guid = Guid.NewGuid();
+                    var slice = span.Slice((GuidTwoGiBLimit + 1));
+                    slice[0] = guid;
+                    slice = span.Slice(GuidOneGiBLimit).Slice(1).Slice(GuidOneGiBLimit);
+                    Assert.Equal(guid, slice[0]);
+                }
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(huge);
+            }
+        }
+
+        [Fact]
+        public static void ArrayCtorStartOverflow()
```

Comment:
How does this name apply to this test? What does "overflow" here mean?

Also, what is this testing that doesn't already have tests?

Reply:
This may have been a redundant test.